app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,770 @@
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with And Bible.
+ * If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+package net.bible.service.db.readingplan
+
+import android.content.ContentValues
+import android.provider.BaseColumns
+import android.util.Log
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.JSON
+import net.bible.android.BibleApplication
+import net.bible.android.activity.R
+import net.bible.android.view.activity.readingplan.DailyReading
+import net.bible.service.common.CommonUtils
+import net.bible.service.db.CommonDatabaseHelper
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_FALSE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_TRUE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanDays
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanMeta
+import net.bible.service.readingplan.PassageReader
+import org.apache.commons.lang3.StringUtils
+import org.crosswire.jsword.passage.Key
+import org.crosswire.jsword.versification.Versification
+import org.crosswire.jsword.versification.system.SystemNRSVA
+import org.crosswire.jsword.versification.system.Versifications
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import kotlin.collections.ArrayList
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/14/2019)
+ */
+class ReadingPlanDBAdapter {
+    companion object {
+        const val TAG = "ReadingPlanDBAdapter"
+        const val INCLUSIVE_VERSIFICATION = SystemNRSVA.V11N_NAME
+        val app = BibleApplication.application
+        val dbHelper: CommonDatabaseHelper = CommonDatabaseHelper.getInstance()
+
+        private val dbWritable = dbHelper.writableDatabase
+        private val dbReadable = dbHelper.readableDatabase
+
+        private val dateBasedFormatMonthDay = SimpleDateFormat("MMM-d")
+        val dateBasedFormatWithYear = SimpleDateFormat("MMM-d/yyyy")
+
+    }
+
+    var metaCurrentActiveReadingPlanID: Int?
+        get() {
+            val projection = arrayOf(BaseColumns._ID)
+            val selection = "${ReadingPlanMeta.COLUMN_LAST_USED_PLAN}=?"
+            val selectionArgs = arrayOf(DB_TRUE_VALUE)
+            val cursor = dbReadable.query(
+                ReadingPlanMeta.TABLE_NAME,
+                projection,
+                selection,
+                selectionArgs,
+                null,
+                null,
+                null)
```

Comment:
You could consider putting those arguments inline without creating separate variables

Reply:
Done